### PR TITLE
Metal compute support

### LINKED ISSUE #1743: Compute on Metal
Are there any blockers now w.r.t. compute support?

https://github.com/KhronosGroup/SPIRV-Cross/pull/321 has been merged a while ago, plus we got https://github.com/gfx-rs/metal-rs/pull/22 . cc @grovesNL

### PR DESCRIPTION
**Pull request by kvark**
Fixes #1743
Depends on https://github.com/gfx-rs/metal-rs/pull/27
Includes part of #1751
r? @grovesNL

### CODE REVIEW

File: src/backend/metal/src/command.rs [hunk 1]
```diff
@@ -125,23 +129,39 @@ impl CommandBufferInner {
 
         self.resources_vs.clear();
         self.resources_fs.clear();
+        self.resources_cs.clear();
+    }
+
+    fn stop(&mut self) {
```
Comment (msiglreith) [inline src/backend/metal/src/command.rs]: Could this be called `stop_encoding` or something similar?

File: src/backend/metal/src/device.rs [hunk 2]
```diff
@@ -133,19 +112,47 @@ pub struct Device {
...
 impl PhysicalDevice {
+    pub(crate) fn new(raw: metal::Device) -> Self {
+        PhysicalDevice {
+            raw,
+            memory_types: [
+                hal::MemoryType {
+                    properties: Properties::CPU_VISIBLE | Properties::CPU_CACHED,
+                    heap_index: 0,
+                },
+                hal::MemoryType {
```
Comment (msiglreith) [inline src/backend/metal/src/device.rs]: why is a second memory type here, exposing the same things as the first one?

Comment (kvark) [reply src/backend/metal/src/device.rs]: this was just copied from the previous code. Perhaps @grovesNL remembers?

Comment (grovesNL) [reply src/backend/metal/src/device.rs]: It was added in #1687 (https://github.com/gfx-rs/gfx/pull/1687/files#diff-97544b45a9d25f924c3a661a46b1cd0fR63). I guess it's not important unless there is some special meaning for these four indices?

Comment (grovesNL) [reply src/backend/metal/src/device.rs]: It doesn't seem to be important unless those indices have special meaning elsewhere? These properties became the same in #1571 with the removal of the `memory::WRITE_COMBINED` (https://github.com/gfx-rs/gfx/pull/1571/files#diff-97544b45a9d25f924c3a661a46b1cd0fL120)

**Comment by kvark** [comment]
Requested fixes are applied. Linux and Windows are covered by CI. OSX is covered by compiling locally.
Merge time!

**Pull request merged**